### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by dosvath**
This resolves issue #1 
Added unit tests for grpcat.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,6 +21,32 @@ Assumptions
...
+
+Given a protobuf definition, `hello.proto`, compile it to python:
+
+`$ python3 -m  grpc_tools.protoc  -I. --python_out=. --grpc_python_out=. hello.proto`
+
+Create a `get_hello_request.txt` file with the appropriate message field specified in `hello.proto`, for example `name:"Joe"`. 
+
+Run the following way:
+
+`$ cat get_hello_request.txt | unit_tests.py`
```
Comment (adamf) [inline README.md]: We shouldn't require the user to create the file to run the unit tests. You should instead:

Create a hello.proto:
```
package grpcat;

message Hello {
  required string name = 1;
}
```

Compile it, and add the proto file and the resulting generated module to this branch. 

Then, in order to call grpcat as a module inside the unit tests, you would use a mock of fileinput.input to set standard in to the string that would be in get_hello_request.txt. See python 3's `unittest` and `mock` modules (and the `@patch` decorator) to see how to do this. This way the user doesn't have to supply the file to the unit test.

File: unit_tests.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+        args = grpcat.parse_args()
+        try:
+            grpcat.run(args)
+        except Exception as e:
+            self.assertEqual(fail_message, str(e))
+
+
+
+    def test_successful_connection(self):
```
Comment (adamf) [inline unit_tests.py]: How will this test ever pass if there's no server?

File: unit_tests.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        for i in range(len(test_args)):
+            #define sys args
+            sys.argv[1:] = test_args[i]
+            #get return from the arg parse function
+            retval = grpcat.parse_args()
+            #assertEqual
+            self.assertEqual(expected_parse[i], str(retval))
+
+
+    def test_arg_count_errors(self):
```
Comment (adamf) [inline unit_tests.py]: This test fails for me:

```
======================================================================
FAIL: test_arg_count_errors (__main__.TestArgumentParsing)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "unit_tests.py", line 86, in test_arg_count_errors
    self.assertEqual(err.getvalue(),expected_errors[i])
AssertionError: u'' != 'usage: unit_tests.py [-h] host port rpc message_type\nunit_tests.py: error: the following arguments are required: message_type\n'
```

File: unit_tests.py (hunk 2 above)
Comment (dosvath) [reply unit_tests.py]: The test checks whether grpcat raises any exceptions, if there are no exceptions raised by grpcat, there should be a response from the server. We could add the mock server for testing, or the test could be there for testing once there is a server?

File: unit_tests.py (hunk 3 above)
Comment (dosvath) [reply unit_tests.py]: Did you run using python3? It runs on my side...

Comment (adamf) [reply unit_tests.py]: Ah, you are right - sorry!

File: unit_tests.py (hunk 2 above)
Comment (adamf) [reply unit_tests.py]: We should mock the server. It'll be better in the long run.

Comment (dosvath) [reply unit_tests.py]: For the mock server would it be better to create a `mock_server.py` or start it on a thread in the `unit_tests.py`?

File: README.md (hunk 1 above)
Comment (dosvath) [reply README.md]: In the new commit I created the hello.proto and its compiled modules, implemented the mock fileinput.input, and created a sample get_hello_request.txt

**Review by adamf — COMMENTED** [review]
Please run unit_tests.py to through a PEP-8 formatter; after that this will look good to commit. We'll have some post-commit issues to open, but I don't want to delay this landing.

**Pull request merged**